### PR TITLE
Allow using default expression in alter column

### PR DESCRIPTION
**Pull request by git-hulk**

### CODE REVIEW

**Review by Joker-hh — APPROVED** [review]

File: pkg/ast/ast.go [hunk 1]
```diff
@@ -1584,9 +1584,10 @@ type AlterColumn struct {
 	Alter token.Pos // position of "ALTER" keyword
 	Null  token.Pos // position of "NULL"
 
-	Name    *Ident
-	Type    SchemaType
-	NotNull bool
+	Name        *Ident
+	Type        SchemaType
+	NotNull     bool
+	DefaultExpr *ColumnDefaultExpr
```
Comment (makenowjust) [inline pkg/ast/ast.go:1590]: Please update the above doc comment, and `End` method.

File: pkg/ast/ast.go [hunk 2]
```diff
@@ -1598,8 +1599,9 @@ type AlterColumnSet struct {
 
 	Alter token.Pos // position of "ALTER" keyword
 
-	Name    *Ident
-	Options *ColumnDefOptions
+	Name        *Ident
+	Options     *ColumnDefOptions
+	DefaultExpr *ColumnDefaultExpr
```
Comment (makenowjust) [inline pkg/ast/ast.go:1604]: Ditto.

**Review by makenowjust — CHANGES_REQUESTED** [review]
Thank you!

To update doc comment and `End` method is missing. Please do that.

File: pkg/ast/ast.go (hunk 1 above)
Comment (git-hulk) [reply pkg/ast/ast.go:1590]: Sure, my bad.

File: pkg/ast/ast.go [hunk 3]
```diff
@@ -1576,11 +1576,10 @@ type SetOnDelete struct {
 
 // AlterColumn is ALTER COLUMN clause in ALTER TABLE.
 //
-//     ALTER COLUMN {{.Name | sql}} {{.Type | sql}} {{if .NotNull}}NOT NULL{{end}}
+//     ALTER COLUMN {{.Name | sql}} {{.Type | sql}} {{if .NotNull}}NOT NULL{{end}} {{.DefaultExpr | sql}}
```
Comment (makenowjust) [inline pkg/ast/ast.go:1579]: It should be `{{.DefaultExpr | sqlOpt}}` instead.

File: pkg/ast/ast.go [hunk 4]
```diff
@@ -1592,10 +1591,10 @@ type AlterColumn struct {
 
 // AlterColumnSet is ALTER COLUMN SET clause in ALTER TABLE.
 //
-//     ALTER COLUMN {{.Name | sql}} SET {{.Options | sql}}
+//     ALTER COLUMN {{.Name | sql}} SET {{.Options | sql}} || {{.DefaultExpr | sql}}
```
Comment (makenowjust) [inline pkg/ast/ast.go:1594]: It should be `{{if .Options}}{{.Options | sql}}{{else}}{{.DefaultExpr | sql}}{{end}}`.

**Review by makenowjust — CHANGES_REQUESTED** [review]

File: pkg/ast/ast.go (hunk 4 above)
Comment (git-hulk) [reply pkg/ast/ast.go]: Fixed, sorry that I didn't understand the doc syntax exactly. Any guideline for this?

**Review by makenowjust — APPROVED** [review]

File: pkg/ast/ast.go (hunk 4 above)
Comment (makenowjust) [reply pkg/ast/ast.go]: Syntax is Go's `text/template`, and the three extra filters `sql`, `sqlOpt`, and `sqlJoin` are defined.

- `sql` calls `SQL` method and returns its result.
- `sqlOpt` is similar to `sql`, but it returns empty string when the input is `nil`.
- `sqlJoin` calls `SQL` method against each element of the input slice, then returns the string joining results with the argument as its separator.

**Comment by git-hulk** [comment]
@makenowjust Should I add more test case for `End` function to make codecov happy?

**Comment by makenowjust** [comment]
@git-hulk It sounds good.

**Pull request merged**